### PR TITLE
fix(IT Wallet): [SIW-779,SIW-781] Add proximity saga

### PR DESCRIPTION
**Pull request by hevelius**
## Short description
This PR adds proximity saga handlers.

NOTE: this is an experimental proposal about proximity flow using redux-saga and could be change in future

## List of changes proposed in this pull request
- Add itwProximitySaga

## How to test
Static checks should be enough

### CODE REVIEW

File: ts/features/it-wallet/saga/itwProximitySaga.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    yield* put(stopProximityManager.success(true));
+  } catch {
+    yield* put(stopProximityManager.failure(new Error("Stop failed")));
+  }
+  yield* put(
+    proximityManagerStatus({
+      status: ProximityManagerStatusEnum.STOPPED
+    })
+  );
+}
```
Comment (LazyAfternoons) [inline ts/features/it-wallet/saga/itwProximitySaga.ts:74]: Should we put this in the try block? 
Also, we are not using `ItWallerErrors`, is this intended?

**Review by LazyAfternoons — APPROVED** [review]

File: ts/features/it-wallet/saga/itwProximitySaga.ts (hunk 1 above)
Comment (hevelius) [reply ts/features/it-wallet/saga/itwProximitySaga.ts:74]: This action is only meant to track the status of the proximity manager and must be stopped in any case. For ItWalletError I was thinking about a next refactor. I don't know what type of error would be useful to map. Maybe I can use a GENERIC. What do you think?

Comment (hevelius) [reply ts/features/it-wallet/saga/itwProximitySaga.ts:74]: I'm adding a generic proximity error type just now....

Comment (hevelius) [reply ts/features/it-wallet/saga/itwProximitySaga.ts:87]: https://github.com/pagopa/io-app/pull/5360/commits/49d115537d0cfa890fe08d3a845b4f27ccdcd4b1

**Review by LazyAfternoons — APPROVED** [review]
LGTM

**Pull request merged**